week-1-ts-ecosystem/w01e04-light-me-up/src/main.ts

```diff
@@ -8,14 +8,82 @@
  * Dodatkowo — rzucać błąd, jeśli zapas mocy się wyczerpie.
  * */
 
+const MILLISECONDS_IN_SECOND = 1000;
+
 class PowerSource {
-  private energySupply = 100
+  private energySupply = 100 // J
+
+  public consume(energy: number) {
+    const rest = this.energySupply - energy;
+
+    if (rest < 0) {
+      throw Error("Not enough energy");
+    }
 
-  consume(energy) {
-    this.energySupply -= energy
+    console.log(`Consume ${energy}J. Remaining energy: ` + rest + "J");
+    this.energySupply = rest;
   }
 }
 
 class LightBulb {
-  protected readonly powerConsumption = 20
+  protected readonly powerConsumption = 20 // W
+  private powerSource: PowerSource;
+  private consumption: number = -1;
```

Comment:
Dałbym tutaj: `private consumption = -1`, bez `:number` konsekwentnie tak jak wyżej.

Co do samej nazwy `consumption` nie jestem przekonany bo to pole sterujące `ID` naszego interwału.
Może jakaś bliższa nazwa faktycznnej odpowiedzialności tego `consumption` - co myślisz?